################################################################################
SHORT DESCRIPTION:
################################################################################
Always used saved connection profiles. If none exist, let networkmanager create
default ones.

################################################################################
LONG DESCRIPTION:
################################################################################
Now that networkmanager actively manages wired/wireless/and brbridged devices+
connections, if we make changes to these connections and save them to disk (db)
for future use, due to networkmanager's core logic, it creates new connections
with the same ID but refuses to attempt to try to match them. This causes
multiple connections with the same IDs to be shown in the nm-applet.

The keyfile plugin we use to save/parse connection profiles will read and 
generate connections before NetworkManager runs through its core autogeneration
logic. This patch removes a flag that is set always set to FALSE during this 
initialization and lets NetworkManager attempt to match on an existing connection
profile.

This allows connection setting changes, such as DNS entries or password credentials,
to persist across ndvm and host platform reboots.

################################################################################
CHANGELOG
################################################################################
Authors:
Chris Rogers <rogersc@ainfosec.com>

################################################################################
REMOVAL
################################################################################
None

################################################################################
UPSTREAM PLAN
################################################################################
None

################################################################################
INTERNAL DEPENDENCIES
################################################################################
db-nm-settings.patch

################################################################################
PATCHES
################################################################################
--- a/src/nm-manager.c
+++ b/src/nm-manager.c
@@ -2580,7 +2580,7 @@ get_existing_connection (NMManager *self
 		return NULL;
 	}
 
-	if (!matched && assume_state_guess_assume) {
+	if (!matched) {
 		gs_free NMSettingsConnection **sett_conns = NULL;
 		guint len, i, j;
 
